x/committee/types/committee.go

```diff
@@ -73,32 +121,121 @@ func (c Committee) Validate() error {
-	}
-
 	for _, p := range c.Permissions {
 		if p == nil {
 			return fmt.Errorf("committee cannot have a nil permission")
 		}
 	}
 
+	if c.ProposalDuration < 0 {
+		return fmt.Errorf("invalid proposal duration: %s", c.ProposalDuration)
+	}
+
 	// threshold must be in the range (0,1]
 	if c.VoteThreshold.IsNil() || c.VoteThreshold.LTE(sdk.ZeroDec()) || c.VoteThreshold.GT(sdk.NewDec(1)) {
 		return fmt.Errorf("invalid threshold: %s", c.VoteThreshold)
 	}
 
-	if c.ProposalDuration < 0 {
-		return fmt.Errorf("invalid proposal duration: %s", c.ProposalDuration)
+	return nil
+}
+
+// String implements fmt.Stringer
+func (c BaseCommittee) String() string {
+	return fmt.Sprintf(`Committee %d:
+  Description:              %s
+  Members:               %s
+  Permissions:               			%s
+  VoteThreshold:            		  %s
+  ProposalDuration:        						%s
+  TallyOption:   						%d`,
+		c.ID, c.Description, c.GetMembers(), c.Permissions,
+		c.VoteThreshold.String(), c.ProposalDuration.String(),
+		c.TallyOption,
+	)
+}
+
+// MemberCommittee is an alias of BaseCommittee
+type MemberCommittee struct {
+	BaseCommittee `json:"base_committee" yaml:"base_committee"`
+}
+
+// NewMemberCommittee instantiates a new instance of MemberCommittee
+func NewMemberCommittee(id uint64, description string, members []sdk.AccAddress, permissions []Permission,
+	threshold sdk.Dec, duration time.Duration, tallyOption TallyOption) MemberCommittee {
+	return MemberCommittee{
+		BaseCommittee: BaseCommittee{
+			ID:               id,
+			Description:      description,
+			Members:          members,
+			Permissions:      permissions,
+			VoteThreshold:    threshold,
+			ProposalDuration: duration,
+			TallyOption:      tallyOption,
+		},
+	}
+}
+
+// GetType returns the type of the committee
+func (c MemberCommittee) GetType() string { return MemberCommitteeType }
+
+// TokenCommittee supports voting on proposals by token holders
+type TokenCommittee struct {
+	BaseCommittee `json:"base_committee" yaml:"base_committee"`
+	Quorum        sdk.Dec `json:"quorum" yaml:"quorum"`
+	TallyDenom    string  `json:"tally_denom" yaml:"tally_denom"`
+}
+
+// NewTokenCommittee instantiates a new instance of TokenCommittee
+func NewTokenCommittee(id uint64, description string, permissions []Permission, threshold sdk.Dec,
+	duration time.Duration, tallyOption TallyOption, quorum sdk.Dec, tallyDenom string) TokenCommittee {
+	return TokenCommittee{
+		BaseCommittee: BaseCommittee{
+			ID:               id,
+			Description:      description,
+			Permissions:      permissions,
+			VoteThreshold:    threshold,
+			ProposalDuration: duration,
+			TallyOption:      tallyOption,
+		},
+		Quorum:     quorum,
+		TallyDenom: tallyDenom,
+	}
+}
+
+// GetType returns the type of the committee
+func (c TokenCommittee) GetType() string { return TokenCommitteeType }
+
+// Validate validates the committee's fields
+func (c TokenCommittee) Validate() error {
```

Comment:
This method should return `c.BaseCommittee.Validate()`

Reply:
https://github.com/Kava-Labs/kava/pull/899/commits/5ff4eeb6591dcf2040bb900af67fdfe6b2165c98